src/policy/semantic.rs

```diff
@@ -14,6 +14,7 @@
 
 //! Abstract Policies
 
+use std::collections::HashSet;
```

Comment:
prefer using BTreeSet since it gives you reproducible results, although less critical for abstract policies than concrete.